File: orders-service/handler.js
/**
 * Form Submit
 */

const https = require('https');
var AWS = require('aws-sdk');
AWS.config.update({region:'us-east-1'});

var sns = new AWS.SNS();
var dynamodb = new AWS.DynamoDB();

const submit = async (event, context) => {
  context.serverlessSdk.tagEvent('customerId', 5, { newCustomer: true, isDemo: true, demoExpire: '2019-12-01' })
  if (event['queryStringParameters'] && event['queryStringParameters']['error']) {
    let r = Math.random().toString(36).substring(7);
    throw new Error(`Random error ${r}`)
  }
  
  // Create and delete an SNS Topic
  try {
    var createParams = {
      Name: "slspartnertopic1"
    }
    var createResult = await sns.createTopic(createParams).promise()
    console.log(createResult)
    var deleteParams = {
      TopicArn: createResult["TopicArn"]
    }
    var deleteResult = await sns.deleteTopic(deleteParams).promise()
    console.log(deleteResult)
  } catch (e) {
    console.log(e, e.stack)
  }

  // Make HTTP request
  await new Promise((resolve, reject) => {
    const req = https.get({ host: 'www.nokeynoshade.party', path: '/api/queens/89/', method: 'get' }, resp => {
      let data = '';
      resp.on('data', chunk => {
        data += chunk;
      });
      resp.on('end', () => {
        resolve(data);
      });
    });
    req.on('error', reject);
  });

  // Send items to DynamoDB
  try {
    const dynamoParams = {
      Item: {
        "email": { S: "fernando@serverless.com" }, 
      }, 
      ReturnConsumedCapacity: "TOTAL", 
      TableName: "orders-table"
    };
    const putData = await dynamodb.putItem(dynamoParams).promise()
    console.log('Put the item in DynamoDB')
  } catch (e) {
    console.log(e, e.stack)
  }

  const response = {
    statusCode: 200,
    headers: {
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Credentials': true,
    },
    body: JSON.stringify({ message: 'form submission received' }),
  }

  return response
}

module.exports = { submit }

File: orders-service/handler.py
import boto3
import requests
import random
import json

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('orders-table')
sns = boto3.client('sns')

def process(event, context):
    # Set a random user for this transaction
    # All of this would normally come from event data
    user = random.choice(['alice', 'bob', 'charlie', 'david', 'eric', 'fernando'])
    user_type = random.choice(['paid', 'free'])
    context.serverless_sdk.tag_event('customerId', 5, { 'newCustomer': True, 'isDemo': True})
    context.serverless_sdk.tag_event('plan', user_type, {'optional-data': 'included with the tag', 'guid': '1237r512tye7r561ft7r5ds6a'})
    context.serverless_sdk.tag_event('user', user, {'optional-data': 'included with the tag', 'guid': '1237r512tye7r561ft7r5ds6a'})
    # Randomly throw errors
    if random.choice([True, False]):
        print("This is where a line of code will break")
        ses.send_email(Email="Stuff that's not gonna send")
        raise Exception('Whoops we made an error if we even get this far')

    # Create and delete topics
    topic = sns.create_topic(Name='slspartnertopic2')['TopicArn']
    print("Topic created: " + topic)
    sns.delete_topic(TopicArn=topic)
    print("Topic deleted: " + topic)

    # HTTP request
    print("requesting nokeynoshade data")
    r = requests.get("http://www.nokeynoshade.party/api/queens/89")
    print(r.status_code)
    print("Data:")
    print(r.text)

    # Send item into DynamoDB Table
    response = table.put_item(
        Item={'email': 'anotheremail@gmail.com'}
    )
    print("Table Item Added")
    return {
        "statusCode": 200,
        "body": json.dumps(response)
    }
